src/vs/workbench/contrib/chat/common/chatAgents.ts

```diff
@@ -311,15 +327,23 @@ export class ChatAgentService extends Disposable implements IChatAgentService {
 	private _updateContextKeys(): void {
 		let editingAgentRegistered = false;
 		let defaultAgentRegistered = false;
+		let toolsAgentRegistered = false;
 		for (const agent of this.getAgents()) {
 			if (agent.isDefault && agent.locations.includes(ChatAgentLocation.EditingSession)) {
 				editingAgentRegistered = true;
+				if (agent.isToolsAgent) {
+					toolsAgentRegistered = true;
+				}
 			} else if (agent.isDefault) {
 				defaultAgentRegistered = true;
 			}
 		}
 		this._editingAgentRegistered.set(editingAgentRegistered);
 		this._defaultAgentRegistered.set(defaultAgentRegistered);
+		if (toolsAgentRegistered !== this._hasAgentContextKey.get()) {
+			this._hasAgentContextKey.set(toolsAgentRegistered);
```

Comment:
Could this be more specific e.g.

```suggestion
			this._hasToolsAgentContextKey.set(toolsAgentRegistered);
```

Reply:
Yeah this might prompt me to finally rename all the "agent" -> "participant"